app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt

```diff
@@ -0,0 +1,159 @@
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.rememberAsyncImagePainter
+import com.android.voyageur.R
+import com.android.voyageur.model.trip.Trip
+import com.android.voyageur.model.trip.TripsViewModel
+import com.android.voyageur.model.user.UserViewModel
+import com.android.voyageur.ui.overview.DisplayParticipants
+
+@Composable
+fun DiscoverContent(
+    tripsViewModel: TripsViewModel,
+    userViewModel: UserViewModel,
+    modifier: Modifier = Modifier
+) {
+  val userId = userViewModel.user.collectAsState().value?.id
+  // Fetch trips
+  tripsViewModel.getFeed(userId ?: "")
+  val trips = tripsViewModel.feed.collectAsState(initial = emptyList()).value
+  val pagerState = rememberPagerState(pageCount = { trips.size })
+  rememberCoroutineScope()
+
+  if (trips.isEmpty()) {
+    NoTripsFound()
+  } else {
+    HorizontalPager(state = pagerState, modifier = modifier.fillMaxSize().testTag("pager")) { page
+      ->
+      TripCard(trip = trips[page], userViewModel = userViewModel)
+    }
+  }
+}
+
+@Composable
+fun TripCard(trip: Trip, userViewModel: UserViewModel) {
+  Box(
+      modifier =
+          Modifier.fillMaxSize()
+              .background(MaterialTheme.colorScheme.background)
+              .padding(16.dp)
+              .testTag("tripCard_${trip.id}"),
+      contentAlignment = Alignment.Center) {
+        Column(
+            modifier =
+                Modifier.fillMaxWidth(0.9f)
+                    .fillMaxHeight()
+                    .clip(RoundedCornerShape(16.dp))
+                    .border(2.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(16.dp))
+                    .background(MaterialTheme.colorScheme.surface)
+                    .padding(16.dp)) {
+              // Trip Image
+              if (trip.imageUri.isNotEmpty()) {
+                Image(
+                    painter = rememberAsyncImagePainter(trip.imageUri),
+                    contentDescription = "Trip Image",
+                    contentScale = ContentScale.Crop,
+                    modifier =
+                        Modifier.fillMaxWidth().height(200.dp).clip(RoundedCornerShape(8.dp)))
+              } else {
+                Image(
+                    painter = rememberAsyncImagePainter(model = R.drawable.default_trip_image),
+                    contentDescription = "Trip Image",
+                    contentScale = ContentScale.Crop,
```

Comment:
For the contentDescription in the if else structure you can consider naming them with something like: "Image of ${trip.name}", for better accuracy. Also, I recommend adding testTags to all the components in order to facilitate the testing.